tests/ethpm/tools/test_builder.py

```diff
@@ -263,6 +288,40 @@ def test_builder_with_standard_token_manifest(
         pin_source("StandardToken", compiler_output, ipfs_backend),
         pin_source("Token", compiler_output, ipfs_backend),
         contract_type("StandardToken", compiler_output, abi=True, natspec=True),
+        validate(),
     )
+    assert manifest == expected_manifest
 
+
+def test_builder_with_link_references(
+    registry_package, dummy_ipfs_backend, monkeypatch
+):
+    root, expected_manifest, compiler_output = registry_package
+    ipfs_backend = get_ipfs_backend()
+    monkeypatch.chdir(str(root))
+    manifest = build(
+        {},
+        package_name("registry"),
+        manifest_version("2"),
+        version("1.0.0"),
+        pin_source("Authority", compiler_output, ipfs_backend),
+        pin_source("IndexedOrderedSetLib", compiler_output, ipfs_backend),
+        pin_source("PackageDB", compiler_output, ipfs_backend),
+        pin_source("PackageIndex", compiler_output, ipfs_backend),
+        pin_source("PackageIndexInterface", compiler_output, ipfs_backend),
+        pin_source("ReleaseDB", compiler_output, ipfs_backend),
+        pin_source("ReleaseValidator", compiler_output, ipfs_backend),
+        pin_source("SemVersionLib", compiler_output, ipfs_backend),
+        contract_type("Authorized", compiler_output, abi=True),
```

Comment:
Same thought here for reducing boilerplate.

```python
contract_types(
    "Authorized", "PackageDB",
    compiler_output=compiler_output, abi=True, ...
)
# or
extractor = contract_types(compiler_output)
build(
    extractor("PackageDB", abi=True, deployment_bytecode=True)
```

Just some thoughts on how this API could shape up.